dataprofiler/profilers/profile_builder.py

```diff
@@ -1582,8 +1589,21 @@ def __add__(  # type: ignore[override]
             merged_profile.row_is_null_count = (
                 self.row_is_null_count + other.row_is_null_count
             )
-            merged_profile.hashed_row_dict.update(self.hashed_row_dict)
-            merged_profile.hashed_row_dict.update(other.hashed_row_dict)
+
+            if (
+                self.options.row_statistics.unique_count.is_enabled
+                and other.options.row_statistics.unique_count.is_enabled
+            ):
+                if isinstance(self.hashed_row_object, dict) and isinstance(
+                    other.hashed_row_object, dict
+                ):
+                    merged_profile.hashed_row_object.update(self.hashed_row_object)
+                    merged_profile.hashed_row_object.update(other.hashed_row_object)
+                elif isinstance(self.hashed_row_object, HyperLogLog) and isinstance(
+                    other.hashed_row_object, HyperLogLog
+                ):
+                    self.hashed_row_object.merge(other.hashed_row_object),
+                    merged_profile.hashed_row_object = self.hashed_row_object
```

Comment:
Similarly here, if they disagree, we currently continue and can silently report false information. An option is to continue, but disable and clear the results maybe with a warning. Or instead to just raise an error. Either way, test conditions are necessary for these cases.